Salesforce kamelet - Adjust for Camel 3.19.x

The source must be changed to use subscribe as operationName as that is now mandatory in 3.19 onwards.

Otherwise you get errors like

Caused by: org.apache.camel.ResolveEndpointFailedException: Failed to resolve endpoint: local-salesforce-1://OpportunityTopic?notifyForFields=ALL&notifyForOperationCreate=true&notifyForOperationDelete=false&notifyForOperationUndelete=false&notifyForOperationUpdate=false&sObjectQuery=SELECT+Id%2C+name%2C+accountId%2C+stagename%2C+amount++FROM+Opportunity+WHERE+IsWon+%3D+true&updateTopic=true due to: OpportunityTopic
	at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:985)
	at org.apache.camel.impl.engine.AbstractCamelContext.getEndpoint(AbstractCamelContext.java:867)
	at org.apache.camel.support.CamelContextHelper.getMandatoryEndpoint(CamelContextHelper.java:58)
	at org.apache.camel.reifier.AbstractReifier.resolveEndpoint(AbstractReifier.java:191)
	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:94)
	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:74)
	... 25 more
Caused by: java.lang.IllegalArgumentException: OpportunityTopic
	at org.apache.camel.component.salesforce.internal.OperationName.fromValue(OperationName.java:128)
	at org.apache.camel.component.salesforce.SalesforceComponent.createEndpoint(SalesforceComponent.java:304)
	at org.apache.camel.support.DefaultComponent.createEndpoint(DefaultComponent.java:171)
	at org.apache.camel.impl.engine.AbstractCamelContext.doGetEndpoint(AbstractCamelContext.java:951)

--- a/kamelets/salesforce-source.kamelet.yaml
+++ b/kamelets/salesforce-source.kamelet.yaml
@@ -117,7 +117,7 @@ spec:
           - 'urn:alm:descriptor:com.tectonic.ui:checkbox'
         default: false
       operation:
-	title: Operation
+        title: Operation
         description: The operation to use
         type: string
         default: subscribe

--- a/kamelets/salesforce-source.kamelet.yaml
+++ b/kamelets/salesforce-source.kamelet.yaml
@@ -116,6 +116,11 @@ spec:
         x-descriptors:
           - 'urn:alm:descriptor:com.tectonic.ui:checkbox'
         default: false
+      operation:
+	title: Operation
+        description: The operation to use
+        type: string
+        default: subscribe
   types:
     out:
       mediaType: application/json
@@ -143,6 +148,7 @@ spec:
         notifyForOperationDelete: "{{notifyForOperationDelete}}"
         notifyForOperationUndelete: "{{notifyForOperationUndelete}}"
         sObjectQuery: "{{query}}"
+        operationName: "{{operation}}"
       steps:
       - marshal:
           json: {}

--- a/library/camel-kamelets/src/main/resources/kamelets/salesforce-source.kamelet.yaml
+++ b/library/camel-kamelets/src/main/resources/kamelets/salesforce-source.kamelet.yaml
@@ -117,7 +117,7 @@ spec:
           - 'urn:alm:descriptor:com.tectonic.ui:checkbox'
         default: false
       operation:
-	title: Operation
+        title: Operation
         description: The operation to use
         type: string
         default: subscribe

--- a/library/camel-kamelets/src/main/resources/kamelets/salesforce-source.kamelet.yaml
+++ b/library/camel-kamelets/src/main/resources/kamelets/salesforce-source.kamelet.yaml
@@ -116,6 +116,11 @@ spec:
         x-descriptors:
           - 'urn:alm:descriptor:com.tectonic.ui:checkbox'
         default: false
+      operation:
+	title: Operation
+        description: The operation to use
+        type: string
+        default: subscribe
   types:
     out:
       mediaType: application/json
@@ -143,6 +148,7 @@ spec:
         notifyForOperationDelete: "{{notifyForOperationDelete}}"
         notifyForOperationUndelete: "{{notifyForOperationUndelete}}"
         sObjectQuery: "{{query}}"
+        operationName: "{{operation}}"
       steps:
       - marshal:
           json: {}

